src/module_wrap.cc

```diff
@@ -502,6 +502,44 @@ void ModuleWrap::Resolve(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(result.ToObject(env));
 }
 
+static MaybeLocal<Promise> ImportModuleDynamically(Local<Context> context,
+                                                   Local<String> referrer,
+                                                   Local<String> specifier) {
+  Isolate* iso = context->GetIsolate();
+  // TODO(jkrems): ensure proper context is used
+  Environment* env = Environment::GetCurrent(context);
```

Comment:
if we could perform the check that only the main context (`env.context_`) gets `import()` handling setup, that would be good. We don't have a public exposed way in `vm` currently to setup things for creating new contexts.

Reply:
I made it an explicit error condition when someone tries to use dynamic import from another context. Also added some test cases for it.